pandas/tests/io/test_common.py

```diff
@@ -411,3 +413,297 @@ def test_is_fsspec_url():
     assert not icom.is_fsspec_url("random:pandas/somethingelse.com")
     assert not icom.is_fsspec_url("/local/path")
     assert not icom.is_fsspec_url("relative/local/path")
+
+
+def test_plain_text_read_csv_http_custom_headers(monkeypatch):
+    true_df = pd.DataFrame({"column_name": ["column_value"]})
+    df_csv_bytes = true_df.to_csv(index=False).encode("utf-8")
+    headers = {
+        "User-Agent": "custom",
+        "Auth": "other_custom",
+    }
+
+    class DummyResponse:
+        headers = {
+            "Content-Type": "text/csv",
+        }
+
+        @staticmethod
+        def read():
+            return df_csv_bytes
+
+        @staticmethod
+        def close():
+            pass
```

Comment:
This class looks similar in all cases but one.
Can it be parametrized?

```
class DummyResponse:
    def __init__(self, data):
        self.data = data

    def read(self):
        return self.data
```

Then only function ``dummy_response_getter`` should be updated accordingly.

Reply:
My bad, I noticed that ``headers`` class attribute is different from case to case.
But that _can_ be parametrized too (not that I insist on it).

Reply:
I'll have a go at parameterizing it, getting back to this today.  That said, after Jeff's comment on preferring not to mock (in combination with not being sure how to use monkeypatch in a way that legitimately tests this functionality) I think I will rewrite the tests to use threading and a simple http server like I had in a past commit.  I bailed on that in favor of patching because I seemed to need a port per test and wasn't sure if I could rely on ports not being taken.  I'm fine with that route though if that's what's preferred, any guidance offered (ports, useful fixtures that already exist, etc.) is well taken.